app/web/modules/gateway_accounts/gateway_accounts.http.js

```diff
@@ -36,31 +40,44 @@ const writeAccount = async function writeAccount (req, res, next) {
   const account = new GatewayAccount(req.body)
   const linkedService = req.body.systemLinkedService
 
-  jobs.account = await Connector.createAccount(account.formatPayload())
-  logger.info(`Created new Gateway Account ${jobs.account.gateway_account_id}`)
+  const createAccountMethod = account.isDirectDebit ? DirectDebitConnector.createAccount : Connector.createAccount
+  jobs.account = await createAccountMethod(account.formatPayload())
+
+  logger.info(`Created new Gateway Account ${jobs.account.gateway_account_id} with external ID ${jobs.account.gateway_account_external_id}`)
+
+  // derive Gateway account ID. For Direct Debit use `gateway_account_external_id` otherwise `gateway_account_id`
+  const gatewayAccountIdDerived = account.isDirectDebit
+    ? `${jobs.account.gateway_account_external_id}`
+    : `${jobs.account.gateway_account_id}`
 
   // connect system linked services to the created account
   if (linkedService) {
-    jobs.linkService = await AdminUsers.updateServiceGatewayAccount(linkedService, jobs.account.gateway_account_id)
-    logger.info(`Service ${linkedService} linked to new Gateway Account ${jobs.account.gateway_account_id}`)
+    jobs.linkService = await AdminUsers.updateServiceGatewayAccount(linkedService, gatewayAccountIdDerived)
+    logger.info(`Service ${linkedService} linked to new Gateway Account ${gatewayAccountIdDerived}`)
 
     jobs.serviceStatus = await AdminUsers.updateServiceGoLiveStatus(linkedService, 'LIVE')
     logger.info(`Service ${linkedService} 'current_go_live_stage' updated to 'LIVE'`)
   }
 
   // note payment_provider is not returned in the object returned from createAccount
-  res.render('gateway_accounts/createSuccess', { account: jobs.account, linkedService, provider: account.provider })
+  res.render('gateway_accounts/createSuccess', { account: jobs.account, linkedService, gatewayAccountIdDerived, provider: account.provider })
 }
 
 const detail = async function detail (req, res, next) {
   const id = req.params.id
   let services = {}
-  const account = await Connector.account(id)
+  let account = ''
+
+  if (id.match(/^DIRECT_DEBIT:/)) {
```

Comment:
This looks like a clean method of determining the appropriate account type 👍  -- it would be good if it could follow the same paradigm set by the another part of this PR https://github.com/alphagov/pay-toolbox/pull/20/files#diff-6508a0aeb9271da11a223a72bfb3954cR43

This avoids state being manipulated and would make it easier to read the code base as a whole.
 
(Or the other method should be updated to look like this one) 

Using the other pattern from this PR might look like:
```js
const isDirectDebitID = id.match(/^DIRECT_DEBIT:/)
const readAccountMethod = isDirectDebitID ? DirectDebitConnecter.account : Connector.account

const account = await readAccountMethod(id)
```

Reply:
Nice!! 👍